src/apps/activities/api/activities.py

```diff
@@ -157,6 +159,62 @@ async def applet_activities_for_subject(
         return Response(result=result)
 
 
+async def applet_activities_for_target_subject(
+    applet_id: uuid.UUID,
+    subject_id: uuid.UUID,
+    user: User = Depends(get_current_user),
+    language: str = Depends(get_language),
+    session=Depends(get_session),
+    answer_session=Depends(get_answer_session),
+) -> ResponseMulti[ActivityOrFlowWithAssignmentsPublic]:
+    applet_service = AppletService(session, user.id)
+    await applet_service.exist_by_id(applet_id)
+
+    await SubjectsService(session, user.id).exist_by_id(subject_id)
+
+    # Restrict the endpoint access to owners, managers, coordinators, and assigned reviewers
+    await CheckAccessService(session, user.id).check_subject_subject_access(applet_id, subject_id)
+
+    assignments = await ActivityAssignmentService(session).get_all_with_subject_entities(
+        applet_id, QueryParams(filters={"target_subject_id": subject_id})
+    )
+
+    activity_and_flow_ids = await AnswerService(
+        session, user.id, answer_session
+    ).get_activity_and_flow_ids_by_target_subject(subject_id)
+
+    activities_and_flows = await ActivityService(session, user.id).get_activity_and_flow_basic_info_by_ids_or_auto(
+        applet_id, activity_and_flow_ids, language
+    )
+
+    result = []
+    for activity_or_flow in activities_and_flows:
+        activity_or_flow_assignments = [
+            assignment
+            for assignment in assignments
+            if assignment.activity_id == activity_or_flow.id or assignment.activity_flow_id == activity_or_flow.id
+        ]
+
+        if activity_or_flow.is_hidden:
+            activity_or_flow.status = ActivityOrFlowStatusEnum.HIDDEN
+        elif len(activity_or_flow_assignments) > 0 or activity_or_flow.auto_assign:
+            activity_or_flow.status = ActivityOrFlowStatusEnum.ACTIVE
+        else:
+            activity_or_flow.status = ActivityOrFlowStatusEnum.INACTIVE
```

Comment:
You can add this code to @validator('status', pre=true) in ActivityOrFlowBasicInfoPublic class, so all other endpoints that may use same return will need to duplicate this

Reply:
Good idea, but I'm not so sure I can since it would need to reference a local variable in the function. Can `activity_or_flow_assignments` be included in the validator?

Reply:
I have the same question as @sultanofcardio. It would be great to be able to produce `status` in the base class so it's available everywhere, but it requires querying assignments – I imagine we shouldn't be making calls to `ActivityAssignmentService` in a validator function?

Reply:
You can implement a @root_validation(pre=true),  the parameter `values` is the dictionary you pass in object construction

Reply:
No you can call any services in any validator

Reply:
Ok I took a look at this and I'm still not seeing a path to the validator. The logic that determines the `status` depends on the list of assignments, which is part of the `ActivityOrFlowWithAssignmentsPublic` class, but that class doesn't have the `is_hidden` property which is also needed to determine the value of the status. To achieve this, I'd need to construct a class that has both these properties and I don't think that's worth the effort.

I think a better outcome would be putting this logic in a static method that we can pass the necessary values to, so it can be reused should the need arise. What do you think about that?

Reply:
This is the code for validator:
`
 class ActivityOrFlowWithAssignmentsPublic(ActivityOrFlowBasicInfoPublic):
    assignments: list[ActivityAssignmentWithSubject] = Field(default_factory=list)

    @root_validator(pre=True)
    def assign_status(cls, values):
        is_hidden = values.get("is_hidden", False)
        assignments = values.get("assignments", [])
        auto_assign = values.get("auto_assign", False)

        if is_hidden:
            values["status"] = ActivityOrFlowStatusEnum.HIDDEN
        elif len(assignments) > 0 or auto_assign:
            values["status"] = ActivityOrFlowStatusEnum.ACTIVE
        else:
            values["status"] = ActivityOrFlowStatusEnum.INACTIVE

        return values
`

and remove this block
<img width="727" alt="image" src="https://github.com/user-attachments/assets/002c4ba7-04c1-490c-9737-cad51a726f28">

Reply:
We chatted about this and decided to forego using the validator this time around because we're not entirely clear on how to make it work in this case. Specifically, the validator relies on the `is_hidden` field, which isn't part of this class. We're gonna move ahead with a member function for now to get this merged